on-screen-keyboard-input/src/KeypadButtons.tsx

```diff
@@ -1,31 +1,42 @@
-import { FC, InputHTMLAttributes } from 'react';
+import { InputHTMLAttributes, MouseEvent } from 'react';
 import { Flex } from './Flex';
 import { KeypadButton } from './KeypadButtons.styled';
 
-export interface IKeypadInputProps extends InputHTMLAttributes<HTMLInputElement> { }
+type IKeypadButtonsProps = InputHTMLAttributes<HTMLInputElement> & {
+	onButtonClick: (str: string) => void;
+}
+
+export const KeypadButtons = ({ onButtonClick, ...props }: IKeypadButtonsProps) => {
+	const handleParentClick = (event: MouseEvent<HTMLDivElement, globalThis.MouseEvent>) => {
+		const target = event.target
+
+		/* make sure we only trigger the onButtonClick event when the target is a button */
+		if (target instanceof HTMLInputElement && target.nodeName === 'INPUT') {
+			onButtonClick(target.value)
+		}
+	}
 
-export const KeypadButtons: FC<IKeypadInputProps> = (props) => {
 	return (
-		<Flex flexDirection="column" gap="4px" width="100%">
+		<Flex flexDirection="column" gap="4px" width="100%" onClick={handleParentClick}>
 			<Flex gap="4px" width="100%">
-				<KeypadButton value="7" />
-				<KeypadButton value="8" />
-				<KeypadButton value="9" />
+				<KeypadButton {...props} value="7" />
```

Comment:
added props drilling to pass the html props to those inputs, because of this comment in a task: `should accept all html input element attributes as props`. However, I'm not sure if we really want to pass those attributes on to all the inputs in both components.
I hope I understood these Acceptance Criteria correctly, but in a real project, of course, I will ask for clarification and the use cases.